contrib/coredns/policy/dns.go

```diff
@@ -114,42 +140,49 @@ func resetTTL(r *dns.Msg) *dns.Msg {
 }
 
 func (p *policyPlugin) setRedirectQueryAnswer(ctx context.Context, w dns.ResponseWriter, r *dns.Msg, dst string) (int, error) {
+	qName, qType, qClass := getNameTypeAndClass(r)
+	rr, err := getRRByType(dst, qName, qType, qClass)
+	if err != nil {
+		return dns.RcodeServerFailure, err
```

Comment:
`getRRByType()` doesn't err if RR is to be got by data format. It only errs if RR is to be got by rrCodePrefix format and an error is encountered in the same only